### PR TITLE
Update default Checkstyle version to latest 8.x release

### LINKED ISSUE #19751: Default checkstyle version does not support sealed modifier
The default version of `checkstyle` bundled with the current (`7.3.3`) distribution does not support Java's `sealed` modifier.

This is outlined in their issue tracker - https://github.com/checkstyle/checkstyle/issues/8989

And included in version `8.42` - https://checkstyle.sourceforge.io/releasenotes.html#Release_8.42

### Expected Behavior
Checkstyle tasks should correctly parse the Java grammar.

### Current Behavior
The build fails with an error. Stacktrace taken from linked project.
```
Execution failed for task ':checkstyleMain'.
> Unable to process files: [{absolute path}\gradle-checkstyle-issue\src\main\java\com\github\scottsteen\SealedClass.java]

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':checkstyleMain'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:145)
	... 32 more
Caused by: : Unable to process files: [{absolute path}\gradle-checkstyle-issue\src\main\java\com\github\scottsteen\SealedClass.java]
	at com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.processFiles(CheckstyleAntTask.java:371)
	at com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.realExecute(CheckstyleAntTask.java:331)
	at com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.execute(CheckstyleAntTask.java:302)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	... 139 more
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing {absolute path}\gradle-checkstyle-issue\src\main\java\com\github\scottsteen\SealedClass.java
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:311)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:221)
	at com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.processFiles(CheckstyleAntTask.java:365)
	... 146 more
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: IllegalStateException occurred while parsing file {absolute path}\gradle-checkstyle-issue\src\main\java\com\github\scottsteen\SealedClass.java.
	at com.puppycrawl.tools.checkstyle.JavaParser.parse(JavaParser.java:120)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:149)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:87)
	at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:337)
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:298)
	... 148 more
Caused by: java.lang.IllegalStateException: {absolute path}\gradle-checkstyle-issue\src\main\java\com\github\scottsteen\SealedClass.java:3:8: unexpected token: sealed
	at com.puppycrawl.tools.checkstyle.JavaParser$1.reportError(JavaParser.java:108)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.typeDefinition(GeneratedJavaRecognizer.java:424)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.compilationUnit(GeneratedJavaRecognizer.java:212)
	at com.puppycrawl.tools.checkstyle.JavaParser.parse(JavaParser.java:114)
	... 152 more
Caused by: {absolute path}\gradle-checkstyle-issue\src\main\java\com\github\scottsteen\SealedClass.java:3:8: unexpected token: sealed
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.typeDefinitionInternal(GeneratedJavaRecognizer.java:596)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.typeDefinition(GeneratedJavaRecognizer.java:402)
	... 154 more
```

### Context
Our dev team uses Checkstyle & Gradle. We've pinned the checkstyle version to the latest `8.x` release, but would prefer to leave the version handling to Gradle.

### Steps to Reproduce 
Example project here -> https://github.com/scottsteen/gradle-checkstyle-issue-reproduction

Running `gradlew checkstyleMain --stacktrace` reproduces the issue.

### Your Environment
Java 17
Gradle 7.3.3

### PR DESCRIPTION
**Pull request by donat**
Fixes #19751

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test RFM

File: subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstylePlugin.java [hunk 1]
```diff
@@ -45,7 +45,7 @@
  */
 public class CheckstylePlugin extends AbstractCodeQualityPlugin<Checkstyle> {
 
-    public static final String DEFAULT_CHECKSTYLE_VERSION = "8.37";
+    public static final String DEFAULT_CHECKSTYLE_VERSION = "8.45.1";
```
Comment (ljacomet) [inline subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstylePlugin.java:48]: Please add a note in the upgrade guide - see the commit that bumped to 8.37 for inspiration.

**Review by ljacomet — CHANGES_REQUESTED** [review]

File: subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstylePlugin.java (hunk 1 above)
Comment (donat) [reply subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstylePlugin.java:48]: Fixed

**Comment by donat** [comment]
@bot-gradle test and merge

**Pull request merged**